src/utils/hypercore.ts

```diff
@@ -0,0 +1,187 @@
+import type { Chain, Network, Signer } from '@wormhole-foundation/sdk-connect';
+import { routes } from '@wormhole-foundation/sdk-connect';
+import type {
+  PermitDomain,
+  PermitValue,
+  PermitTypes,
+} from '@mayanfinance/swap-sdk';
+import { getHyperCoreUSDCDepositPermitParams } from '@mayanfinance/swap-sdk';
+import { isEvmNativeSigner } from '@wormhole-foundation/sdk-evm';
+import type { Eip6963Wallet } from '@wormhole-labs/wallet-aggregator-evm';
+import type {
+  Quote,
+  TransferParams,
+  ValidationResult,
+} from '../routes/mayan/types';
+import { TransferWallet } from 'utils/wallet';
+import type { WormholeConnectWalletProvider } from 'utils/wallet/types';
+import { isEvmChain } from './evm';
+import { isUSDCToken } from './usdc';
+
+type SignerWithProvider<N extends Network> = Signer<N> & {
+  provider?: () => WormholeConnectWalletProvider;
+};
+
+export function isHyperCoreChain(chain: Chain): boolean {
+  return chain === 'HyperCore';
+}
+
+/**
+ * Enforces HyperCore routing rules: only EVM -> HyperCore flows with USDC
+ * inbound, and disallow HyperCore-originated USDC.
+ */
+export function validateHyperCoreTransfer<N extends Network>(
+  request: routes.RouteTransferRequest<N>,
+  params: TransferParams,
+): ValidationResult | null {
+  const { fromChain, toChain, destination } = request;
+
+  if (isHyperCoreChain(fromChain.chain)) {
+    return {
+      valid: false,
+      params,
+      error: new routes.UnavailableError(
+        new Error('HyperCore cannot be used as a source chain'),
+      ),
+    };
+  }
+
+  if (!isHyperCoreChain(toChain.chain)) {
+    return null;
+  }
+
+  if (toChain.network !== 'Mainnet') {
+    return {
+      valid: false,
+      params,
+      error: new routes.UnavailableError(
+        new Error('HyperCore is only available on Mainnet'),
+      ),
+    };
+  }
+
+  if (!isEvmChain(fromChain.chain)) {
```

Comment:
Lets add a comment on top here too that this is temporary as SUI/Solana support is coming soon